CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -293,5 +294,76 @@ def __init__(self, db_client: "botocore.client.DynamoDB"):
 
 
 class Tag(DB_Item):
-    def __init__(self, db_client: "botocore.client.DynamoDB"):
+    REQUIRED_FIELDS = ['tag_name']
+    TABLE_NAME = "tag"
+
+    def __init__(self, db_client: "botocore.client.DynamoDB", tag_name:str= None):
         super().__init__(db_client=db_client)
+        self.tag_name = self.format_tag(tag_name)
+
+    def format_tag(self, tag_name:str) -> str:
+        #Tag_name is alphanumeric, can include hyphen and underscore, with no spaces and all lower cased
+        if tag_name:
+            tag_name = tag_name.lower().replace(" ", "")
+        return tag_name
+
+    @staticmethod
+    def is_valid_tag_name(tag_name:str) -> Tuple[bool, str]:
```

Comment:
I have a suggestion on this method:

* ``'tag_name is None'``, ```tag_name is an empty string```, and `Length of the tag_name is {length_tag_name}. It should be >= 3 but <= 32` can all be under the same error message. They all basically tell the user that the length of the tag name is incorrect. So I think error message `Length of the tag_name is {length_tag_name}. It should be >= 3 but <= 32` is enough. What do you think?
```
is_tag_length_valid = length_tag_name >= MIN_TAG_LENGTH and length_tag_name <= MAX_TAG_LENGTH
if tag_name is None or is_tag_length_valid:
  message = f'Length of the tag_name is {length_tag_name}. It should be >= 3 but <= 32.'
```

Reply:
Good thought about using the constants in the error message - I changed that. 

I changed the error messages (except for None) to be the f string message.  

None and empty string are different types in Python so it will throw an error if I try to get the length of a None type. So, that's why I had that separated out. I will try to include the is_tag_length_valid variable though in the code- I like that.

Reply:
I realized that so I got rid of `is_tag_length_valid` in the example code that I added

Reply:
But up to you :). If you want to include a `None` message I would suggest to make the error message: `Missing Required Field: tag_name`. Though in practice it'll never get there because we check for the required_fields first and that error will be handled there first